src/SFML/Graphics/Shape.cpp

```diff
@@ -192,6 +192,7 @@ void Shape::draw(RenderTarget& target, const RenderStates& states) const
     RenderStates statesCopy(states);
 
     statesCopy.transform *= getTransform();
+    statesCopy.coordinateType = CoordinateType::Pixels;
```

Comment:
Could the state not be overwritten? when calling draw?

Reply:
Hmm there are two ways I can understand your question, but in both cases the answer is no :)
 - If you mean "the state could be overwritten in the `target.draw()` call", then no because `RenderTarget::draw()` takes the state as a const reference, and does not modify it.
 - If you mean "we should not overwrite the state in Shape::draw()", then I'd say we have to, because the interface of `Shape` imposes integer texture coordinates (see `Shape::setTextureRect()`, `Shape::updateTexCoords()`). So the class would break if a user asked for different coordinate type. We could check and assert instead of silently overwrite though, but I'm not sure it's best.

Reply:
What happens if I call `window.draw(shape, sf::CoordinateType::Normalized)`?

Reply:
As currently implemented, this is equivalent to `window.draw(shape)`. It calls `Shape::draw()` which overrides the coordinate type to `Pixels`.

It's probably an error to do this, because `Shape` in its current form cannot work with normalized coordinates. There's nothing fundamental about it, we could update all SFML classes to also support normalized coordinates. I left this out of this PR because there wasn't demand for it (in the issue https://github.com/SFML/SFML/issues/1773 it was made clear that pixel coordinates must remain the default), and also to keep the PR focused on the core capability.

Reply:
Hmm, I guess that makes sense. I wonder though whether we should have an assert if someone does choose to pass normalized as type 🤔

Reply:
Thinking about this some more; I'm not sure the assert or `sf::err` is necessary, because there is already a precedent for silently overwriting the render state without warning. Indeed this is already done today for the texture:
https://github.com/SFML/SFML/blob/1e9161b57e475764e91e1efd9bde80102f16618c/src/SFML/Graphics/Shape.cpp#L197
https://github.com/SFML/SFML/blob/1e9161b57e475764e91e1efd9bde80102f16618c/src/SFML/Graphics/Shape.cpp#L203
https://github.com/SFML/SFML/blob/1e9161b57e475764e91e1efd9bde80102f16618c/src/SFML/Graphics/Sprite.cpp#L138
https://github.com/SFML/SFML/blob/1e9161b57e475764e91e1efd9bde80102f16618c/src/SFML/Graphics/Text.cpp#L374

If someone were to call `window.draw(sprite, someOtherTexture)`, the call would ignore `someOtherTexture` and just use the texture set up inside `sprite`, without warning. So it seems there is already an implicit contract that render states may be overwritten inside `draw()`, if it is necessary to display the `Drawable` as intended.